packages/stack-shared/src/config/parser.ts

```diff
@@ -0,0 +1,163 @@
+// Check out https://github.com/stack-auth/info/blob/main/eng-handbook/random-thoughts/config-json-format.md for more information on the config format
+
+import * as yup from "yup";
+import { yupNumber, yupObject, yupTuple } from "../schema-fields";
+import { deepPlainClone } from "../utils/objects";
+
+export function mergeConfigs(options: {
+  configs: any[],
+  configSchema: yup.AnySchema,
+}) {
+  let mergedConfig: {} | yup.InferType<typeof options.configSchema> = {};
+
+  for (const config of options.configs) {
+    mergedConfig = mergeConfig({
+      configSchema: options.configSchema,
+      defaultConfig: mergedConfig,
+      overrideConfig: config,
+    });
+  }
+
+  return mergedConfig;
+}
+
+export function mergeConfig(options: {
+  configSchema: yup.AnySchema,
+  defaultConfig: any,
+  overrideConfig: any,
+}): yup.InferType<typeof options.configSchema> {
```

Comment:
can you add some validation to this function? in particular:

- throw an error if a key does not follow the format
- throw an error when deeply trying to set a nested key that doesn't exist
- throw an error if the input object is not an object

StackAssertionErrors on all of them